meadow-client/src/Semantics.purs

```diff
@@ -1159,88 +1154,67 @@ applyTransaction inputs sigs blockNum state contract value = case appResult of
+    = S.union (S.fromFoldable (M.keys rpp)) (S.fromFoldable (map (\x -> x.person) (M.values ccbi)))
 
-peopleFromObservation (ValueGE value1 value2) = S.union (peopleFromValue value1) (peopleFromValue value2)
+instance peopleFromState :: PeopleFrom State where
+  peopleFrom (State { commits: comm }) = peopleFrom comm
 
-peopleFromObservation (ValueGT value1 value2) = S.union (peopleFromValue value1) (peopleFromValue value2)
-
-peopleFromObservation (ValueLT value1 value2) = S.union (peopleFromValue value1) (peopleFromValue value2)
-
-peopleFromObservation (ValueLE value1 value2) = S.union (peopleFromValue value1) (peopleFromValue value2)
-
-peopleFromObservation (ValueEQ value1 value2) = S.union (peopleFromValue value1) (peopleFromValue value2)
-
-peopleFromObservation TrueObs = S.empty
-
-peopleFromObservation FalseObs = S.empty
-
-peopleFromContract :: Contract -> S.Set Person
-peopleFromContract Null = S.empty
-
-peopleFromContract (Commit _ _ person value _ _ contract1 contract2) = S.insert person (S.union (peopleFromValue value) (S.union (peopleFromContract contract1) (peopleFromContract contract2)))
-
-peopleFromContract (Pay _ _ person value _ contract1 contract2) = S.insert person (S.union (peopleFromValue value) (S.union (peopleFromContract contract1) (peopleFromContract contract2)))
-
-peopleFromContract (Both contract1 contract2) = S.union (peopleFromContract contract1) (peopleFromContract contract2)
-
-peopleFromContract (Choice observation contract1 contract2) = S.union (peopleFromObservation observation) (S.union (peopleFromContract contract1) (peopleFromContract contract2))
-
-peopleFromContract (When observation timeout contract1 contract2) = S.union (peopleFromObservation observation) (S.union (peopleFromContract contract1) (peopleFromContract contract2))
-
-peopleFromContract (While observation timeout contract1 contract2) = S.union (peopleFromObservation observation) (S.union (peopleFromContract contract1) (peopleFromContract contract2))
-
-peopleFromContract (Scale value1 value2 value3 contract) = S.union (peopleFromValue value1) (S.union (peopleFromValue value2) (S.union (peopleFromValue value3) (peopleFromContract contract)))
-
-peopleFromContract (Let _ contract1 contract2) = S.union (peopleFromContract contract1) (peopleFromContract contract2)
-
-peopleFromContract (Use _) = S.empty
+instance peopleFromValue :: PeopleFrom Value where
+  peopleFrom = cata algebra
+    where
+      algebra :: Algebra ValueF (S.Set Person)
+      algebra CurrentBlockF = S.empty
+      algebra (CommittedF _) = S.empty
+      algebra (ConstantF _) = S.empty
+      algebra (NegValueF value) = value
+      algebra (AddValueF value1 value2) = S.union value1 value2
+      algebra (SubValueF value1 value2) = S.union value1 value2
+      algebra (MulValueF value1 value2) = S.union value1 value2
+      algebra (DivValueF value1 value2 value3) = S.unions [value1, value2, value3]
```

Comment:
`value1 <> value2 <> value3`?

Reply:
I like this style

Reply:
Fair enough.

Reply:
I meant I like your style @krisajenkins !

Reply:
:-D